src/connection/connection-manager.ts

```diff
@@ -0,0 +1,58 @@
+import type { Dwn } from "@tbd54566975/dwn-sdk-js";
+
+import type { IncomingMessage } from "http";
+import type { WebSocket } from 'ws';
+
+import { SocketConnection } from "./socket-connection.js";
+
+/**
+ * Interface for managing `WebSocket` connections as they arrive.
+ */
+export interface ConnectionManager {
+  /** connect handler used for the `WebSockets` `'connection'` event. */
+  connect(socket: WebSocket, request?: IncomingMessage): Promise<void>;
+  /** cleans up handlers associated with the `WebSocket` connection */
+  close(socket:WebSocket): Promise<void>;
+  /** closes all of the connections */
+  closeAll(): Promise<void>
+}
+
+/**
+ * A Simple In Memory ConnectionManager implementation.
+ * It uses a `Map<WebSocket, SocketConnection>` to manage connections.
+ */
+export class InMemoryConnectionManager implements ConnectionManager {
+  constructor(private dwn: Dwn, private connections: Map<WebSocket, SocketConnection> = new Map()) {}
+
+  /**
+   * Handler for opening websocket event - `connection`.
+   * Sets listeners for `message`, `pong`, `close`, and `error` events.
+   */
```

Comment:
Consider removing the comments in this case since they come from the `ConnectionManager` already, more ideal for VSC pick up documentation directly from the interface. Same goes to other interface methods.

Reply:
Done